src/generators.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any, Generator, Iterator
+
+
+def filter_by_currency(transactions: list[dict[str, Any]], currency: str = "USD") -> Iterator[dict[str, Any]]:
+    """Функция перебора списка словарей, она принимает на вход список словарей и возвращает,
+    итератор который выдает транзакции если они соответствуют заданной валюте"""
+
+    return (x for x in transactions if x.get("operationAmount", {}).get("currency", {}).get("name") == currency)
```

Comment:
Не нужно использовать get. Если ключа нет в словаре, значит кривой словарь и ты должен упасть с ошибкой, а не умалчивать ее:
```suggestion
    return (x for x in transactions if x["operationAmount"]["currency"["name"] == currency)
```